File: python-programming/examples/dictionary/count_characters.py
text = """
This is a very long text.
OK, maybe it is not that long after all.
"""

# print text
count = {}
for c in text:
    if c == '\n':
        continue
    if c not in count:
        count[c] = 0
    count[c] += 1

for k in count:
    print("'{}' {}".format(k, count[k]))

File: books/golang/src/examples/count-characters/count_characters.go
package main

import "fmt"

func main() {
	text := "This is a very long text. OK, maybe it is not that long after all."
	counter := make(map[string]int)
	for _, c := range text {
		//fmt.Printf("%v", string(c))
		if string(c) != " " {
			counter[string(c)]++
		}
	}

	fmt.Println(text)
	for char, count := range counter {
		fmt.Printf("%v: %v\n", char, count)
	}

}
